### ISSUE TITLE
Double Blind Spotting with Aerospace Fighters on Ground Map

### ISSUE THREAD

**Issue by ParsonBoyles** [MegaMek/megamek#2457]
I'm not sure if this is a "bug" or just something I don't understand fully, but I was asked on the slack channel to write it up, so here we go! It might be complicated to explain.

I'm using Megamek 0.47.14 on Windows. My Java is the most recent 64 bit, 15.0.1

I was using Aerospace Fighters to scout enemy units in Double Blind rules. I am unsure of how the rules governing fighters in  air-to-ground spotting worked, and after scouring the rule books my reading of the sparse rules doesn't line up with the way they are implemented in the megamek. If the answer to all my questions is a simple reference to a section in a rule book that I haven't found, I'll be very happy. 

I have three main interrelated issues:

1) How do active and passive ASF air-to-air sensors work and where are their ranges detailed?
2) What are the rules governing air-to-ground sensors?
3) Why is air-to-air visual spotting, air-to-ground visual spotting and ground-to-air visual spotting all so different?

I'll try to explain further.

The only reference I can find to ASFs in Double Blind rules is on page 222 of TacOps, which basically says sensor rules for ASFs are the same as Mechs and ASFs have LOS to "all units on the map." Plus the Maximum Visual Range chart on Page 221.

In MegaMek, ASFs have two sets of sensors, passive and active. Passive radar has very long ranges (I saw one that was 147 hexes) and active radar has much shorter ranges (like 10 hexes) that don't seem to match up with the chart of sensors on page 222 of TacOps. What are passive and active sensors? Are the active and passive sensors pulling from the Advanced Sensors section in StratOps pg 119? In which case, they refer to space hexes, not the map hexes used by ground units. So... yeah, i'm confused. :)

The Aerospace Fighter units in the Megamek also have air-to-ground sensors ranges, which I can't find any reference to in the rule books. Is the extremely short ranges of the air-to-ground sensors (max of 3 hexes, I think) taken from any source? It seems to rule them out as airborne recon units in a Double Blind game.

Aside from sensors, the other confusing issue is visual spotting. The visual spotting distance (120 hexes in daylight) only seems to apply to air-to-sir situations. When it comes to air-to-ground visual spotting, the ASFs seem to only "see" ground units directly below their flight path. BUT ground units can "see" air units if the air units' ground path is within their visual spotting distance (60 hexes in daylight). This leads to some odd moments, like an entire lance blasting away at an airborne unit that can't see any of the units firing at it.

It gets even stranger because if an ASF lands, it becomes a ground unit and can suddenly "see" other ground units around it, units that it couldn't see from the air a moment ago.

Here's some screenshots that illustrate some of these points:

![Fighterspotting_Team1View](https://user-images.githubusercontent.com/64654277/100684071-9f1b6780-3347-11eb-84df-cf2330a57234.png)

Here we have the view from a bunch of mechs on the ground. They can clearly see both ASFs on the other team, the Blackwasp (directly overhead) and the Cheetah.

![FighterSpotting_team2Blackwasp](https://user-images.githubusercontent.com/64654277/100684180-d1c56000-3347-11eb-8918-52d9f8537da1.png)

Here we have the same scene but from the view of the Aerospace Fighters on Team 2. None of the mechs are visible, even though the Blackwasp is directly overhead of one. (I assume it's because this is just after the Deployment phase, before any movement.) Three mechs one hex away from the Blackwasp are spotted only as sensor contacts from the air-to-ground sensor range of one. None of the other mechs which are two and three hexes away from the Blackwasp are spotted visually either from the Blackwasp or the nearby Cheetah. 

I have an almost identical scenario with landed ASFs.

![FighterspottingMap2_Team2Grounded](https://user-images.githubusercontent.com/64654277/100684463-629c3b80-3348-11eb-97d3-108325742a26.png)

The mechs and ASFs are in an almost identical deployment, except the fighters are on the ground. This is from the perspective of Team 2, and all of the enemy mechs are clearly visually spotted.

Here is also a save game file in case it is helpful.

[FighterSpotting_1.sav.gz](https://github.com/MegaMek/megamek/files/5619637/FighterSpotting_1.sav.gz)

Team 1 is user name: Scott with no password. Team 2 is user name: Team 2 with no password.

I hope that I've been clear enough with this long post! Thanks for all your awesome work and for reading through all this.

**Comment by HammerGS** [comment]
Can you please confirm what edition of TacOps you are using should be in the credits section.

**Comment by ParsonBoyles** [comment]
The copyright date is 2008-2010. Is that the edition information? Or is there another number I'm missing?

**Comment by HammerGS** [comment]
Yes, that is an older version. It's had a lot of errata released since then.  Here is a link to the latest errata of that version - https://bg.battletech.com/errata/

TacOps has also been updated and split into two books - links presented as an FYI only
https://store.catalystgamelabs.com/collections/battletech/products/battletech-tactical-operations-advanced-rules

and

https://store.catalystgamelabs.com/collections/battletech/products/battletech-tactical-operations-advanced-units-equipement

You raise very valid issues with the RAW with CGL, and has been raised with them here - https://bg.battletech.com/forums/tactical-operations/research-aerospace-visual-spotting-in-double-blind-games/

**Comment by ParsonBoyles** [comment]
Oh, wow! Okay, thanks. I forgot to scrub through the errata. It's hard to keep all the sources of rules straight.

Thanks so much for clarifying this for me. Does the newest TacOps book address the passive and active sensors for fighters? That part I don't see in the errata.

I appreciate you clarifying this for me. I'll try adding to the forum post you directed me to.

**Comment by HammerGS** [comment]
Ok, the top my head it doesn't.  You won't be able to add to that forum post, but feel free to open another rules query about it.  The more questions answered the more the rules are reviewed.

**Comment by ParsonBoyles** [comment]
I was just realizing that I couldn't bump that post. Haha!

Will do!

**Comment by HammerGS** [comment]
If you do post questions make sure to include a link to that post, but only if your question would link to it.

**Comment by HammerGS** [comment]
CGL has updated with Errata
https://bg.battletech.com/forums/index.php?topic=69700.msg2024543#new
